Tests/Controller/AzineEmailControllerTest.php

```diff
@@ -0,0 +1,190 @@
+<?php
+namespace Azine\EmailBundle\Tests\Controller;
+use Azine\EmailBundle\DependencyInjection\AzineEmailExtension;
+use Azine\EmailBundle\Services\AzineEmailTwigExtension;
+use Azine\EmailBundle\Services\Pagination;
+use Azine\EmailBundle\Tests\FindInFileUtil;
+use Azine\EmailBundle\Services\AzineTemplateProvider;
+use Azine\EmailBundle\Entity\SentEmail;
+use Azine\EmailBundle\Controller\AzineEmailTemplateController;
+use Azine\PlatformBundle\Services\EmailTemplateProvider;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\File\Exception\FileNotFoundException;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\HttpFoundation\Session\Storage\MockFileSessionStorage;
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Security\Core\Authentication\Token\UsernamePasswordToken;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\DomCrawler\Crawler;
+use Symfony\Bundle\FrameworkBundle\Client;
+use Azine\EmailBundle\Tests\TestHelper;
+use Doctrine\ORM\EntityManager;
+
+class AzineEmailControllerTest extends WebTestCase
+{
+    public function testAdminEmailsDashboardAction()
+    {
+        $this->checkApplication();
+
+        // Create a new client to browse the application
+        $client = static::createClient();
+        $client->followRedirects();
+
+        $manager = $this->getEntityManager();
+        $sentEmailRep = $manager->getRepository("Azine\EmailBundle\Entity\SentEmail");
+
+        $sentEmails = $sentEmailRep->findAll();
+
+        $count = count($sentEmails);
+
+        $pagination = $this->getContainer()->get('azine.email.bundle.pagination');
+
+        $manager = $this->getEntityManager();
+
+        // make sure there is some data in the application
+        if ($count < $pagination->getDefaultPageSize()) {
+
+            $amountToAdd = $pagination->getDefaultPageSize() * 2;
+            TestHelper::addSentEmails($manager, $amountToAdd);
+
+            $count += $amountToAdd;
+        }
+
+        $listUrl = substr($this->getRouter()->generate("admin_emails_dashboard", array('_locale' => "en")), 13);
```

Comment:
Why 13?
substr('....',13) ?

Reply:
This is for removing (app_test.php) from url. I took this from Azine\MailgunWebhooksBundle\Tests\Controller\MailgunEventControllerTest  line 116 .